osu.Game.Rulesets.Osu/Difficulty/Evaluators/AimEvaluator.cs

```diff
@@ -33,8 +39,46 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, bool with
             var osuLastObj = (OsuDifficultyHitObject)current.Previous(0);
             var osuLastLastObj = (OsuDifficultyHitObject)current.Previous(1);
 
+            // Doubletap detection stuff
+
+            // Calculate how much things are overlapping. Low overlapping circles are very hard in being doubletappable
+            double overlapnessCurr = Math.Clamp(osuCurrObj.LazyJumpDistance / OsuDifficultyHitObject.NORMALISED_RADIUS / 2, 0, 1);
+            overlapnessCurr = CosSigmoid(overlapnessCurr);
+            overlapnessCurr *= overlapnessCurr;
+
+            double antiOverlapnessLast = 1 - Math.Clamp(osuLastObj.LazyJumpDistance / OsuDifficultyHitObject.NORMALISED_RADIUS / 2, 0, 1);
+            antiOverlapnessLast = CosSigmoid(antiOverlapnessLast);
+            antiOverlapnessLast *= antiOverlapnessLast;
+
+            // Doubletap hitwindow lands in range [0, 300 hitwindow], where 0 is impossible to doubletap 
+            double doubletapHitWindow = (osuLastObj.HitWindowGreat + osuLastLastObj.HitWindowGreat) / 2 - osuLastObj.StrainTime;
+            
+            // Extra time to aim when doubletapping
+            double doubletapTime = 0;
+
+            if (doubletapHitWindow > 0)
+            {
+                // Always equal or bigger than doubletapTime from `else`
+                doubletapTime = osuLastObj.StrainTime * overlapnessCurr * antiOverlapnessLast / 2;
+            }
+            else 
+            {
+                doubletapTime = osuLastObj.HitWindowGreat * overlapnessCurr * antiOverlapnessLast / 2;
+
+                // if the second note is delayed - decrease the penalty down to 0 on doubled time
+                double difference = osuCurrObj.DeltaTime - osuLastObj.DeltaTime;
+                double min = Math.Min(osuCurrObj.DeltaTime, osuLastObj.DeltaTime);
+                difference = Math.Clamp(difference, -min, min);
+
+                doubletapTime *= 1 - CosSigmoid(difference, min);
+            }
+
+
+            osuCurrObj.AdjustedStrainTime = osuCurrObj.StrainTime;
+            if (nerfDoubletap) osuCurrObj.AdjustedStrainTime += doubletapTime;
```

Comment:
I think you're missing a variable here

Reply:
I removed it and forgot to remove if from here, already fixed